src/Microsoft.Health.Dicom.Core/Features/Store/StoreDatasetValidator.cs

```diff
@@ -216,14 +227,14 @@ string EnsureRequiredTagIsPresent(DicomTag dicomTag)
             // add to stack to keep iterating when SQ type, otherwise validate
             foreach (DicomItem item in ds)
             {
-                if (item.ValueRepresentation == DicomVR.SQ)
+                if (item is DicomSequence)
```

Comment:
nit: you can cast it inline like:
```suggestion
                if (item is DicomSequence sequence)
```